pdc_client/__init__.py

```diff
@@ -161,7 +258,13 @@ def decode(req):
 
         content_type = "application/json"
         encode = json.dumps
-        self.client = beanbag.BeanBag(url, session=self.session, fmt=(content_type, encode, decode))
+        # In get_paged function it returns an interator, so user can get each pages quickly in plugins.
+        # The original call function of BeanBag returns the list.
+        # Keep the current codes and options in codes. Just init MultiPageBeanBag when is_all_page is true.
+        if is_all_page:
+            self.client = MultiPageBeanBag(url, session=self.session, fmt=(content_type, encode, decode))
+        else:
+            self.client = BeanBag(url, session=self.session, fmt=(content_type, encode, decode))
```

Comment:
It would be nice to emit some deprecation warning for the BeanBag path, so that at some point it can be removed.

Reply:
Actually I think we just use these two branch, if user wants to get all data, just set is_all_page to true and use MultiPageBeanBag, otherwise just use the original codes to BeanBag